### PR TITLE
rpm: use rpm.execute() instead of fork() and execp() for trigger scriplets

### PR DESCRIPTION
**Pull request by tpgxyz**
Hi,
i suggest to use `rpm.execute()`.

For example:
`assert(rpm.execute("/foo/bar"))`
is equivalent to:
```
pid = posix.fork()
if pid == 0 then
  assert(posix.exec("/foo/bar"))
elseif pid > 0 then
  posix.wait(pid)
end
```

`rpm.execute()` was added in rpm-4.15.0 on 2018.

Distrubutions that are still running rpm older than 4.15.0 can use shell version for triggers.

### CODE REVIEW

File: src/rpm/triggers.systemd.in [hunk 1]
```diff
@@ -115,25 +100,13 @@ end
...
 -- This script will automatically update udev with new rules if files
 -- have been installed or updated in {{UDEV_RULES_DIR}}.
-if posix.access("/run/systemd/system") then
-    pid = posix.fork()
-    if pid == 0 then
-        assert(posix.execp("udevadm", "control", "--reload"))
-    elseif pid > 0 then
-        posix.wait(pid)
-    end
+if posix.access("/run/udev/control") then
```
Comment (keszybz) [inline src/rpm/triggers.systemd.in]: Please don't bury unrelated changes like this in refactor commits. I think this is reasonable, but it needs to be a separate commit.

**Review by keszybz — CHANGES_REQUESTED** [review]
Looks reasonable in general.

File: src/rpm/triggers.systemd.in (hunk 1 above)
Comment (tpgxyz) [reply src/rpm/triggers.systemd.in]: @keszybz Well i'd like to aign with shell version of the triggers at one bite
https://github.com/systemd/systemd/blob/01ae74c8c7f364dd3a1d71a43af00c34f531d3f7/src/rpm/triggers.systemd.sh.in#L78

Comment (keszybz) [reply src/rpm/triggers.systemd.in]: That is fine. But it still needs to be a separate commit.

Comment (tpgxyz) [reply src/rpm/triggers.systemd.in]: Sure, that is no problem for me.

**Comment by keszybz** [comment]
OK, so this commit looks fine. But why not add a second commit that fixes the path?

**Comment by keszybz** [comment]
Ah, #23548 has that. Let's close this one then.

**Comment by tpgxyz** [comment]
Thanks, and sorry for the confusion :)
